js/modules/k6/data/data.go

```diff
@@ -90,7 +93,55 @@ func (d *Data) sharedArray(call sobek.ConstructorCall) *sobek.Object {
 	}
 
 	array := d.shared.get(rt, name, fn)
-	return array.wrap(rt).ToObject(rt)
+	return array.Wrap(rt).ToObject(rt)
+}
+
+// Reader is the interface that wraps the basic Read method.
+//
+// The data module Reader interface is implemented by types that can read data from a source, such
+// as a CSV file, etc.
+type Reader interface {
```

Comment:
The interface in stdlib looks like:

```
type Reader interface {
	Read(p [][byte](https://pkg.go.dev/builtin#byte)) (n [int](https://pkg.go.dev/builtin#int), err [error](https://pkg.go.dev/builtin#error))
}
```

which is slightly different, as I guess the expectations are.

Just for context, this was designed like this to avoid extra allocations, following a similar pattern. However, there's probably an alternative approach that could consist on somehow "wrapping" one into the other one, and making this package to receive `io.Reader`.

For me, as-is is fine, but I'd be happy to see suggestions with `io.Reader`.

Reply:
No, I don't want to move to `io.Reader`. I hoped there was an interface avilable for it in `bufio`/`scanner`.

Btw, what about the renaming part `s/Reader/LineReader`?

Reply:
Renamed to `RecordReader` for explicitness (if we implemented support for retrieving data from a JSON array for instance, "records" would make more sense than "line")